File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int[] ans = new int[k];
        for(int num :nums){
            map.put(num,map.getOrDefault(num,0)+1);
        }
        for(int i = 0;i<k;i++){
            int max = -1;
            for(int num: map.keySet()){
                if(map.get(num)>max){
                    max = map.get(num);
                    ans[i] = num;
                }
            }
            map.remove(ans[i]);
        }
        return ans;
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq={}
        heap=[]
        ans=[]
        for i in nums:
            if i in freq:
                freq[i]+=1
            else:
                freq[i]=1
        for num,count in freq.items():
            heapq.heappush(heap,(count,num))
            if len(heap)>k:
                heapq.heappop(heap)
        for count,num in heap:
            ans.append(num)
        return ans

        